source/PlayerInfo.h

```diff
@@ -374,9 +388,11 @@ class PlayerInfo {
 	std::set<const System *> seen;
 	std::set<const System *> visitedSystems;
 	std::set<const Planet *> visitedPlanets;
+	std::set<const Sale<Outfit> *> visitedOutfitters;
```

Comment:
There shall not be a difference between visiting a planet and visiting the outfitter on said planet, in terms of knowledge.

If it is truly a sticking point, and there is overwhelming desire for players to have to click in to the outfitter, then make `visitedPlanets` a map. Planets in the map at all = visited, and the value for their key = visited outfitter.

Reply:
as said, I want to keep the separation.

as for the map, changing visitedPlanets will result with a rather bigger change set which might introduce bugs into a perfectly fine working code, are you sure it is wise?

as for the suggestion, if I understood you correctly:
- class Planet in visitedPlanets map => planet was visited.
- class Planet in visitedPlanets map has Class outfitter => outfitter on that planet was visited.

this means that there isn't and difference between planets with unvisited outfitter and planets with no outfitter.
can that cause issues?

Reply:
I don't imagine so, it's possible to check if the planet has no outfitter by other means anyway.

But, if this is supposed to be "realistic" or whatever, just visiting the outfitter on a planet should not be how you track what outfits the player has seen and knows about: the `Sale`s available at a planet can change and the outfits available in any given `Sale` can also change.

> class Planet in visitedPlanets map has Class outfitter => outfitter on that planet was visited.

I'm not really sure what you're saying here.

Reply:
again, I understand what you guys are saying however I disagree.

tehhowch suggested to change visitedPlanets from set to map. as visitedPlanets is used in multiple places, I fear that apply such change can result with more bugs as using map requires more code changes because it behaves different than set.
that said, if this change is considered a blocker, I'll make it.

Reply:
I don't think using the same structure as for visited planets can work for your purposes here.
Would unvisiting a planet mean I no longer know about any outfits I might have seen there?
I also don't think your original approach of tracking which `Sale<Outfit>`s have been seen works either. You really need to track which outfits have been seen individually.

Reply:
can you explain what do you mean by "unvisiting a planet"?

each time a player enters an outfitter, the outfits there are get scanned, e,g, new outfits will be added. so new outfits will get known.
it doesn't seems realistic to me that if one enters outfitter A and find out that he sells outfit B, however later in the game, that outfitter stopped selling outfit B, thus making outfit B unknown again.

saving Sale<Outfit> is just an implementation, it can be changed to use planets names (outfitters doesn't have different names than the host planet).

the PR hold individual outfit registration (), isn't that what you referring to?
the outfitters list is to make the outfitter's map more realistic.

Reply:
I assume that "unvisiting a planet" means that the "ownership" of the plant is shifting?
if so, we can "unknown" the outfitter.
as of the how to discover, I propose a solution, add a preference option called "Outfitter discovery method", the default will be "on landing", second option will be "on enter"
only issue with this is that I need help changing the UI.

thoughts?

Reply:
By 'Unvisit', warp is likely referring to `unvisit <planet>` and `unvisit <system>`
https://github.com/endless-sky/endless-sky/wiki/CreatingEvents#basic-event-characteristics
You can see some examples of it in use in the FW, Remnant and Wanderer storyline. 

I also third that you should not have to explicitly visit the outfitter to know about the outfits there.

Reply:
> By 'Unvisit', warp is likely referring to `unvisit <planet>` and `unvisit <system>` https://github.com/endless-sky/endless-sky/wiki/CreatingEvents#basic-event-characteristics You can see some examples of it in use in the FW, Remnant and Wanderer storyline.
> 

I think I was right in my assumption, e.g. the republic <=> free world allegiance change or the unfettered hai taking over Wanderer worlds

> I also third that you should not have to explicitly visit the outfitter to know about the outfits there.

that is why I offered the preferences solution, it passes the decision to the player, the default keeps the current behavior.

Reply:
> There shall not be a difference between visiting a planet and visiting the outfitter on said planet, in terms of knowledge.
> 
> If it is truly a sticking point, and there is overwhelming desire for players to have to click in to the outfitter, then make `visitedPlanets` a map. Planets in the map at all = visited, and the value for their key = visited outfitter.

while working on other feature, I've encountered the reason why your suggestion is correct, I'll adjust the code accordingly

Reply:
done